src/game.cpp

```diff
@@ -7685,8 +7685,12 @@ void game::examine( const tripoint &examp )
                 return;
             }
         }
+        if( mon != nullptr && mon->has_flag( MF_MILKABLE ) ) {
+            add_msg( m_info, _( "This is a milkable %s." ), mon->name().c_str());
+            iexamine::milk_source( mon );
```

Comment:
Another minor reason why `milk_source` shouldn't be in `iexamine`: you want to return non-`void` here, because you want the examination to stop if player chooses to milk.
With your current version, you could examine a cow that stands on a trap (say, a mine), start milking the cow, then immediately also try to defuse the mine and blow the cow up.

Would be better if you integrated it with `pet_menu` above. Changed `pet_menu` to `monster_menu`, allowed pet effects only if the monster is a pet, but allowed milking if it is milkable. That way pet cows wouldn't require two menus, and I assume later on only pet cows will be milkable anyway.